mitodl/mitxonline
# Adds program support to the dashboard

(draft - realized that there were a couple bugs while writing this PR)

#### Pre-Flight checklist


#### What are the relevant tickets?

#725

#### What's this PR do?

Adds support for displaying program enrollments on the dashboard, including enrollments that are implicit. Program enrollments display the program itself with the featured image of the first course, and learners can click to expand a drawer that shows their enrollment within courses in the program.

#### How should this be manually tested?

Enroll in a course that is part of a program. The program should appear on your dashboard. The course should _not_ appear. If your course enrollment is an audit enrollment, the program should be tagged as "Enrolled in free course". If the enrollment is a verified enrollment, the program will have the "Enrolled in certificate course" tag. 

Add another enrollment in a course in the same program. The drawer and course count should reflect the correct number of enrollments. The tags should follow the above test, but if you have an enrollment of each type (i.e. one audit and one verified), you should see both tags. 

Enroll directly within a program (you will have to create a Program Enrollment record manually to do this). The program enrollment should show on the dashboard but it should list no courses, be tagged as "Not enrolled in any courses", and you should get the normal "Enroll Now" card when expanding the drawer.  

Enroll in a course that isn't assigned to a program. The course enrollment should appear on the dashboard as normal.

Modals for course enrollments should work properly regardless of their location (on the dashboard or in the drawer). 

#### Screenshots (if appropriate)
![Screen Shot 2022-08-02 at 8 10 54 AM](https://user-images.githubusercontent.com/945611/182382961-e5b015a9-d92e-4bd6-ae63-01db52b17139.png)
![Screen Shot 2022-08-02 at 8 11 09 AM](https://user-images.githubusercontent.com/945611/182383000-e8e9b08a-5543-4f38-b4a5-50f9f7b51fc2.png)
![Screen Shot 2022-08-02 at 8 11 16 AM](https://user-images.githubusercontent.com/945611/182383026-349357a6-9508-4432-b5b2-f8eb6bac1f3c.png)

Width for the drawer is 1280px/701px = 55%; can be changed if necessary.
